api-report/datastore-definitions.api.md

```diff
@@ -124,7 +124,7 @@ export interface IFluidDataStoreRuntime extends IFluidRouter, IEventProvider<IFl
     // (undocumented)
     readonly rootRoutingContext: IFluidHandleContext;
     submitSignal(type: string, content: any): void;
-    uploadBlob(blob: ArrayBufferLike): Promise<IFluidHandle<ArrayBufferLike>>;
+    uploadBlob(blob: ArrayBufferLike, signal?: AbortSignal): Promise<IFluidHandle<ArrayBufferLike>>;
```

Comment:
I like the direction, but I think we need to have a discussion if AbortSignal is sufficient.

I believe most scenarios like that would want to get feedback about situations like "your call will not complete for next 10 minutes because service has high load" (translating 429 with retryAfter). I can totally see where app can in-place progress indication and tooltip with this kind of message.

Another example (in some future, when our interaction and API support it) - 50% of 10Mb were already uploaded.